### PR TITLE
[OAP-CACHE] Update travis v2 branch 0.8 spark 2.4.x -test

### PR DESCRIPTION
**Pull request by offthewall123**
## What changes were proposed in this pull request?

(Please fill in changes proposed in this fix)

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

File: oap-cache/oap/src/main/scala/org/apache/spark/sql/execution/datasources/oap/filecache/OapCache.scala [hunk 1]
```diff
@@ -986,15 +986,15 @@ class ExternalCache(fiberType: FiberType) extends OapCache with Logging {
 
   override def getFibers: Set[FiberId] = {
     //        TODO: Pass CI
```
Comment (winningsix) [inline oap-cache/oap/src/main/scala/org/apache/spark/sql/execution/datasources/oap/filecache/OapCache.scala:988]: Remove this TODO please.

File: oap-cache/oap/src/main/scala/org/apache/spark/sql/execution/datasources/oap/filecache/OapCache.scala [hunk 2]
```diff
@@ -967,8 +967,8 @@ class ExternalCache(fiberType: FiberType) extends OapCache with Logging {
       val plasmaClient = plasmaClientPool(clientRoundRobin.getAndAdd(1) % clientPoolSize)
       try {
       //  TODO: Pass CI
-      //  val buf = plasmaClient.create(objectId, fiber.size().toInt)
-        val buf: ByteBuffer = null
+        val buf = plasmaClient.create(objectId, fiber.size().toInt)
+      //  val buf: ByteBuffer = null
```
Comment (winningsix) [inline oap-cache/oap/src/main/scala/org/apache/spark/sql/execution/datasources/oap/filecache/OapCache.scala:971]: Remove this comment out.

File: oap-cache/oap/src/main/scala/org/apache/spark/sql/execution/datasources/oap/filecache/OapCache.scala [hunk 3]
```diff
@@ -967,8 +967,8 @@ class ExternalCache(fiberType: FiberType) extends OapCache with Logging {
       val plasmaClient = plasmaClientPool(clientRoundRobin.getAndAdd(1) % clientPoolSize)
       try {
       //  TODO: Pass CI
```
Comment (winningsix) [inline oap-cache/oap/src/main/scala/org/apache/spark/sql/execution/datasources/oap/filecache/OapCache.scala:969]: Remove this comment.

File: oap-cache/oap/src/main/scala/org/apache/spark/sql/execution/datasources/oap/filecache/OapCache.scala [hunk 4]
```diff
@@ -929,9 +929,9 @@ class ExternalCache(fiberType: FiberType) extends OapCache with Logging {
       try{
         logDebug(s"Cache hit, get from external cache.")
         val plasmaClient = plasmaClientPool(clientRoundRobin.getAndAdd(1) % clientPoolSize)
-//        TODO: PASS CI
-          val buf: ByteBuffer = null
-//        val buf: ByteBuffer = plasmaClient.getByteBuffer(objectId, -1, false)
+        //      TODO: PASS CI
```
Comment (winningsix) [inline oap-cache/oap/src/main/scala/org/apache/spark/sql/execution/datasources/oap/filecache/OapCache.scala:932]: Remove this TODO.

File: oap-cache/oap/src/main/scala/org/apache/spark/sql/execution/datasources/oap/filecache/OapCache.scala [hunk 5]
```diff
@@ -929,9 +929,9 @@ class ExternalCache(fiberType: FiberType) extends OapCache with Logging {
...
+        //      val buf: ByteBuffer = null
```
Comment (winningsix) [inline oap-cache/oap/src/main/scala/org/apache/spark/sql/execution/datasources/oap/filecache/OapCache.scala:933]: Remove this comment out.

File: .travis.yml [hunk 6]
```diff
@@ -25,7 +53,39 @@ jobs:
...
+      - pushd vmemcache
+      - mkdir build
+      - cd build
+      - cmake .. -DCMAKE_INSTALL_PREFIX=/usr -DCPACK_GENERATOR=deb
+      - make package
+      - sudo dpkg -i libvmemcache*.deb
+      - popd
+      - #install arrow and plasms
+      - cd /tmp
+      - git clone https://github.com/jikunshang/arrow-1.git
```
Comment (winningsix) [inline .travis.yml:76]: Please add TODO here to change to intel-bigdata one.

File: .travis.yml [hunk 7]
```diff
@@ -5,11 +5,39 @@ jobs:
...
```
Comment (winningsix) [inline .travis.yml:28]: Add TODO here.

File: oap-cache/oap/src/main/scala/org/apache/spark/sql/execution/datasources/oap/filecache/OapCache.scala (hunk 1 above)
Comment (offthewall123) [reply oap-cache/oap/src/main/scala/org/apache/spark/sql/execution/datasources/oap/filecache/OapCache.scala]: Done

**Comment by zhixingheyi-tian** [comment]
@offthewall123 @jikunshang 

Can refer for adjust travis:
https://github.com/zhixingheyi-tian/OAP-T-Travis/tree/travis-arrow-shuffle

**Comment by zhixingheyi-tian** [comment]
Add module name to  PR title

File: oap-cache/oap/dev/install_arrow.sh [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+mkdir build
+cd build
+cmake .. -DCMAKE_INSTALL_PREFIX=/usr -DCPACK_GENERATOR=deb
+make package
+sudo dpkg -i libvmemcache*.deb
+popd
+#install arrow and plasms
+cd /tmp
+# TODO change to Intel-bigdata one
+git clone https://github.com/jikunshang/arrow-1.git
```
Comment (zhixingheyi-tian) [inline oap-cache/oap/dev/install_arrow.sh:19]: Is there a bigdata repo?

Comment (offthewall123) [reply oap-cache/oap/dev/install_arrow.sh:19]: this is kunshang'repo,will change to Intel-Bigdata's arrow when merged

File: .travis.yml [hunk 9]
```diff
@@ -7,25 +7,26 @@ jobs:
...
-      - make
-      - sudo make install
-      - cd ${TRAVIS_BUILD_DIR}
       script:
       - cd ${TRAVIS_BUILD_DIR}
-      - mvn clean -q -Ppersistent-memory test
+      - mvn clean -q -Ppersistent-memory -Pvmemcache test
     - name: oap-shuffle-remote-shuffle
+      install:
+      - #echo install step
```
Comment (winningsix) [inline .travis.yml:29]: We should not touch shuffle module.

Comment (winningsix) [reply .travis.yml:29]: @gczsjdy

Comment (gczsjdy) [reply .travis.yml:29]: Right~

File: .travis.yml [hunk 10]
```diff
@@ -7,25 +7,26 @@ jobs:
...
     - name: oap-cache-oap
       before_install:
-      - sudo apt-get install cmake
+#      - sudo apt-get update
       - sudo apt-get install libpthread-stubs0-dev
       - sudo apt-get install libnuma-dev
+      - echo $TRAVIS_COMMIT_MESSAGE
+      - sudo apt-get install cmake
+      - cd ${TRAVIS_BUILD_DIR}/oap-cache/oap/dev
+      - ./install_vmemcache.sh
```
Comment (zhixingheyi-tian) [inline .travis.yml:16]: How about moving ${TRAVIS_BUILD_DIR}/oap-cache/oap/dev/*sh to ${TRAVIS_BUILD_DIR}/dev

Comment (zhixingheyi-tian) [reply .travis.yml:16]: And execute these shell scripts in "before_script" section

File: .travis.yml [hunk 11]
```diff
@@ -7,25 +7,26 @@ jobs:
     #oap-cache-oap is a CI building demo of the corresponding module oap-cache/oap.
     - name: oap-cache-oap
       before_install:
-      - sudo apt-get install cmake
+#      - sudo apt-get update
```
Comment (winningsix) [inline .travis.yml:10]: Remove this please.

Comment (offthewall123) [reply .travis.yml]: done

File: .travis.yml (hunk 10 above)
Comment (offthewall123) [reply .travis.yml]: done

File: dev/install_arrow.sh [hunk 12]
```diff
@@ -0,0 +1,15 @@
+#install arrow and plasms
+cd /tmp
+# TODO change to Intel-bigdata one
+git clone https://github.com/jikunshang/arrow-1.git
```
Comment (winningsix) [inline dev/install_arrow.sh:4]: Please remove the TODO and use intel-bigdata/Arrow.

File: .travis.yml (hunk 9 above)
Comment (winningsix) [reply .travis.yml:28]: Any updates for this?

File: .travis.yml [hunk 13]
```diff
@@ -7,25 +7,29 @@ jobs:
...
       - cd ${TRAVIS_BUILD_DIR}
-      - mvn clean -q -Ppersistent-memory test
+      - mvn clean -q -Ppersistent-memory -Pvmemcache test
     - name: oap-shuffle-remote-shuffle
+      install:
+      # travis will add a command by default
+      # mvn install -DskipTests=true -Dmaven.javadoc.skip=true -B -V
+      # if we don't define an install step in our job,this command will run before all the commands in our travis.yml and run in ${TRAVIS_BUILD_DIR} directory according to our pom.xml in ${TRAVIS_BUILD_DIR}
+      # better way to define at least one install step and one script step in a job even if there is no actual actions in them
+      - #empty install step
```
Comment (offthewall123) [inline .travis.yml:32]: @gczsjdy @winningsix 
Add comments about why define an empty install step here

Comment (gczsjdy) [reply .travis.yml:32]: Thank you, it seems ok for me.
I suggest that we ask Travis guys if there are other ways to disable running that `mvn install...` command. And hopefully we can eliminate these commented lines in the future.

**Comment by winningsix** [comment]
@offthewall123 which JIRA should I close?

**Comment by offthewall123** [comment]
> @offthewall123 which JIRA should I close?

https://jira.devtools.intel.com/browse/POAE7-250 @winningsix

**Pull request merged**